### PR TITLE
[DependencyInjection] Fail when dumping a Definition with no class nor factory

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q | A |
| --- | --- |
| Bug fix? | yes |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | - |
| License | MIT |
| Doc PR | - |

I tried creating an anonymous inline service with a DefinitionDecorator, but that did not work.
Here is the fix.

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
ping @symfony/deciders

File: src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php [hunk 1]
```diff
@@ -1228,6 +1223,15 @@ private function dumpValue($value, $interpolate = true)
                 }
             }
 
+            $class = $value->getClass();
+            if (null === $class) {
+                throw new RuntimeException('Cannot dump definitions which have a no class nor factory.');
```
Comment (fabpot) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: `...which have no class...`

Comment (nicolas-grekas) [inline src/Symfony/Component/DependencyInjection/Dumper/PhpDumper.php]: thanks, fixed

**Comment by fabpot** [comment]
Thank you @nicolas-grekas.

**Pull request merged**